fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/group/ChunkReader.java

```diff
@@ -144,37 +159,48 @@ private void fillChunkDataBuffer(List<Member> patientRefs) throws Exception {
-                    txn.commit();
-                    compartmentPageNum++;
-
-                    for (Resource res2 : resources2) {
-                        if (res2 == null) {
-                            continue;
-                        }
-                        try {
-                            FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
-                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
-                            resSubTotal++;
-                        } catch (FHIRGeneratorException e) {
-                            logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
-                                        + patientId + "'", e);
-                        } catch (IOException e) {
-                            logger.warning("fillChunkDataBuffer: chunkDataBuffer written error!");
-                            throw e;
-                        }
+
+                for (Member patientRef : patientRefs) {
+                    if (patientRef == null) {
+                        continue;
                     }
 
-                } while (searchContext.getLastPageNumber() >= compartmentPageNum);
-            }
+                    String patientId =  patientRef.getEntity().getReference().getValue().substring(8);
+                    searchContext = SearchUtil.parseQueryParameters("Patient", patientId,
+                            ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, true);
+                    do {
+                        searchContext.setPageSize(pageSize);
+                        searchContext.setPageNumber(compartmentPageNum);
+                        FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                        txn.enroll();
+                        FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                        List<Resource> resources2 = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                        txn.unenroll();
+                        compartmentPageNum++;
+
+                        for (Resource res2 : resources2) {
+                            if (res2 == null) {
+                                continue;
+                            }
+                            try {
+                                FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
+                                chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
+                                resSubTotal++;
+                            } catch (FHIRGeneratorException e) {
+                                logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
```

Comment:
so this ends in a warning... does the whole job fail?

Reply:
just logging and continue.